core/services/document_service.py

```diff
@@ -61,13 +76,15 @@ async def retrieve_chunks(
         k: int = 5,
         min_score: float = 0.0,
         use_reranking: Optional[bool] = None,
+        use_colpali: Optional[bool] = None,
     ) -> List[ChunkResult]:
         """Retrieve relevant chunks."""
         settings = get_settings()
         should_rerank = use_reranking if use_reranking is not None else settings.USE_RERANKING
 
         # Get embedding for query
-        query_embedding = await self.embedding_model.embed_for_query(query)
+        query_embedding_regular = await self.embedding_model.embed_for_query(query)
+        query_embedding_multivector = await self.colpali_embedding_model.embed_for_query(query)
```

Comment:
since we take in a bool. shouldn't we embed this conditionally? We store below only conditionally, so shouldn't waste time in computing the embedding